File: prob-1.java
//TC:O(N)
//SC:O(N)


class Solution {
    public int findJudge(int N, int[][] trust) {
        
       if(N==1) return 1;
        
        HashMap<Integer,Integer> map = new HashMap<>();
        HashSet<Integer> set = new HashSet<>();
        
        for(int i=0;i<trust.length;i++)
            set.add(trust[i][0]);
        
        for(int i=0;i<trust.length;i++)
         {
             map.put(trust[i][1],map.getOrDefault(trust[i][1],0)+1); 
         }
        
        for(int k : map.keySet())
        if((map.get(k)==N-1) && (!(set.contains(k))) )
                 return k;
        
        return -1;
    }
}

File: prob1.py
# Time Complexity : O(n)
# Space Complexity :O(n)
# Passed on Leetcode: yes

class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        arr = [0 for _ in range(n)]

        for val in trust:
            trustedby = val[0]
            trusts = val[1]

            arr[trustedby-1] -= 1
            arr[trusts-1] += 1

        for i in range(n):
            if arr[i] == n-1:
                return i+1
        return -1